### ISSUE TITLE
Weights for frame prediction used in the paper

### ISSUE THREAD

**Issue by bsautermeister** [emansim/unsupervised-videos#16]
Hi,

I'm currently trying to do frame prediction in TensorFlow. Your paper is one of the best I have found so far.
Unfortunately, after I have downloaded your code and ran the training of the combo-1layer model for about 600,000 iterations, it looks like the frame predictor does not seem to learn anything. At least there was not improvement from the 100,000th to the 600,000th iteration, where the cross-entropy loss sticks at a value of about 2,600.

**Is it therefore possible that you upload the weights (h5 file) that was used in the paper as well?**
Because first, the training takes very long, seconds, even after training it on a Titan X for about 3 days (600,000 iterations) I do not get such nice results.

**Comment by emansim** [comment]
Hey

Interesting. Last time I tried running it I had no problem converging and managed to train within a day or two on Titan/Titan Black GPUs.

The UCF-101 patches I uploaded specifically here (the one used in the paper was much larger) is a very small dataset due to memory constraints so as far as I remember I managed to heavily overfit at training set.

I will try finding weights but I can't promise anything :)

**Comment by bsautermeister** [comment]
Thank you for your fast answer!

I just realized I was not that precise.
Up to now I only tried to train it on MovingMNIST. For training data (which is using the ground truth as input to the future-predictor), it is actually converging quite good. I think it was down to a training-loss to about 170, but the validation-loss (which is using the previously predicted frames as  input to the future-predictor) stays at a value of about 2600. In consequece, it can predict the first frame very well, but it does not seem to learn the overall motion that good.

**Comment by emansim** [comment]
Hm, I don't remember having this issue of validation loss being way worse compared to training loss.

**Comment by jbboin** [comment]
Hi,

I’m a bit late to the party but it seems that I found the reason why this problem occurred. It was actually caused by the bug fixing that occurred in commit fae11d6. During testing, for each time step the sigmoid was applied to the inputs of the LSTM (which are the outputs of the previous stages). But in the end, the sigmoid was applied again to all outputs. This caused it to be incorrectly applied twice for the outputs at t = 0, …, T-2. My pull request (#26) fixes this problem.

Running the default training mode (1 layer LSTM) on MovingMNIST now gives me a validation cross entropy of around 335 after 120k iterations, which is much more consistent with the values reported in the paper.

**Comment by emansim** [comment]
Thanks for noticing and fixing this bug @jbboin !
